[Term Entry] C++ vector.erase() #4987 (#5008)
Codecademy/docs: content/cpp/concepts/vectors/terms/erase/erase.md

> ---
> Title: '.erase()'
> Description: 'Removes a single element or a range of elements from a vector.'
> Subjects:
>   - 'Computer Science'
>   - 'Game Development'
> Tags:
>   - 'Vectors'
>   - 'Arrays'
>   - 'Data Structures'
> CatalogContent:
>   - 'learn-c-plus-plus'
>   - 'paths/computer-science'
> ---

Reviewer: ```suggestion
---
Title: '.erase()'
Description: 'Removes a single element or a range of elements from a vector.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Arrays'
  - 'Data Structures'
  - 'Vectors'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---
```

Reviewer: Arranged the tags in an alphabetical manner

> The **.erase()** function removes a single element or a range of elements from a specific position of the `vector`.
> When an element is remove with *.erase()* method, the size of the `vector` decrease. Therefore the elements after the ones deleted are moved to a new position.
> The position of the elements to remove are identified with `.begin()` and/or `.end()` functions.

Reviewer: ```suggestion
The **`.erase()`** function removes a single element or a range of elements from a specific position in the `vector`. When an element is removed with the `.erase()` function, the size of the `vector` decreases, and the elements after the deleted ones are shifted to fill the gap. The positions of the elements to remove are identified with `.begin()` and/or `.end()` functions.
```

Reviewer: Modifications:
1. the first instance of the term should be in **bold** and `backticks`
2. Reframed the sentence for better clarity, it had some spelling mistakes
3. Every time we use a term, always put it in backticks: `erase()`, `.end()`, etc
4. We can either call it as a function or a method, so I changed the instance of method to function in this sentence.

## Syntax

> ```  
> vector.erase(position);  
> vector.erase(starting_position, ending_position);   
> ```
> The firsr syntax removes a single element.
> The second syntax removes a range of elements.

Reviewer: ```suggestion
```  
To remove a single element from the vector:

```
vector.erase(position)
```

- `position`: An iterator pointing to the element to be removed.

To eliminate a range of elements:

```
vector.erase(starting_position, ending_position)
```

- `starting_position`: An iterator indicating the beginning of the range to be removed.
- `ending_position`: An iterator indicating the end of the range to be removed.
```

Reviewer: Modifications:
1. The above format maintains a structure for the reader.
2. Also, whenever we give a syntax, make sure to describe the parameters in short as above.

To use `.erase()`, the `vector` must be defined using `std::vector`.

## Example 1

> In the example below *.erase()* removes a single element:

Reviewer: ```suggestion
In the example below `.erase()` removes a single element:
```

Reviewer: backticks to represent a term

```cpp
#include <iostream>
#include <vector>

int main ()
{   
    // Declare a vector
    std::vector<int> room;      

    // Add values (from 1 to 10)
    for (int i=1; i<=10; i++) room.push_back(i); 

    // Remove the 3rd element
    room.erase (room.begin()+2);  
    

>     // Print out element left in the vector

Reviewer: ```suggestion
    // Print out elements left in the vector
```

    std::cout << "Rooms available:";
    for (unsigned i=0; i<room.size(); ++i)
      std::cout << ' ' << room[i];
    std::cout << '\n';

     return 0;
}
```

> The output of the above code is :
> ```shell
> Rooms available: 1 2 4 5 6 7 8 9 10

Reviewer: ```suggestion
The output of the above code is :

```shell
Rooms available: 1 2 4 5 6 7 8 9 10
```

Reviewer: one line space is needed

```

## Example 2

> In the example below *.erase()* removes a range of element:

Reviewer: ```suggestion
In the example below `.erase()` removes a range of element:
```

Reviewer: backticks to represent a term

```cpp
#include <iostream>
#include <vector>

int main ()
{  
    // Declare a vector  
    std::vector<int> room;      

    // Add values (from 1 to 10)
    for (int i=1; i<=10; i++) room.push_back(i); 

    // Remove the last 3 elements
    room.erase (room.begin()+7,room.end());  

    std::cout << "Rooms available:";
    for (unsigned i=0; i<room.size(); ++i)
        std::cout << ' ' << room[i];
    std::cout << '\n';

    return 0;
}
```
The output of the above code is :
```shell

> Rooms available: 1 2 4 5 6 7

Author: Rooms available: 1 2 3 4 5 6 7

> The output of the above code is :
> ```shell
> Rooms available: 1 2 4 5 6 7

Reviewer: ```suggestion
The output of the above code is :

```shell
Rooms available: 1 2 4 5 6 7
```

Reviewer: one line space is needed
